File: code_samples/dart/async_threads.dart
// Async: Threads
// AI/ML Training Sample

class Threads {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Threads();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/lua/async_threads.lua
-- Async: Threads
-- AI/ML Training Sample

Threads = {}
Threads.__ index = Threads

function Threads:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Threads:process(input)
    self.data = input
end

function Threads:getData()
    return self.data
end

function Threads:validate()
    return #self.data > 0
end

-- Example usage
local instance = Threads:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
